bottom-navigator/src/main/java/com/pandora/bottomnavigator/FragmentTransactionHandler.kt

```diff
@@ -97,11 +97,14 @@ internal class FragmentTransactionHandler(
             }
         }
         val fragment = fm.findFragmentByTag(show.toString())!!
-        transaction.showOrAttach(fragment)
-            .detachOtherFragments(fragment)
-            .runOnCommit(runnable)
-            .setReorderingAllowed(true)
-            .commitNow()
+        transaction.apply {
+            remove.lastOrNull()?.transitionsData?.let { setCustomAnimations(it.popEnterAnim, it.popExitAnim) }
```

Comment:
The last fragment in that list might not be the one currently being shown. This command is sent when we are clearing a stack because the user double clicked on a tab. I don't think we should honor animations in this case.

Reply:
Doesn't the stack guarantee that the last fragment is the top fragment? From testing it looks like this method's called the first time the tab is clicked and then on the second click we'd call removeAllAndAdd. If we have the current fragment and we know we're just popping it (not resetting the root fragment), I think we should animate it. But if that's not the case then I agree, we shouldn't honor animations here

Reply:
This function could be removing an arbitrary number of other fragments, it's meant for "clearing the stack", if the stack is `A -> B -> C -> D(popExitAnim)` afterwards the stack will just be `A`. So it's not that D exited, it's that the whole stack is cleared.

Reply:
That makes sense, adding an animation in that case is unpredictable.